### ISSUE TITLE
not working IOS 13.1

### ISSUE THREAD

**Issue by KyDenZ** [abritopach/capacitor-youtube-player#3]
Hello,

Cordova's plugin: Youtube Video Player no longer works under IOS 13.1, I wanted to switch on this one. I wanted to stay on the one of cordova because we can make a play on a button in the form of item-avatar, and we are not obliged to declare on a div the video.

I wish to spend your plugin and see what is possible, but according to the documentation, when I run the application on the simulator, I open the page with the video, I have an error message:

Thread 2: signal SIGABRT

Ionic CLI : 5.2.7
@ionic/angular 4.9.0
NodeJS : v10.15.3
npm : 6.4.1
OS : macOS Mojave
Capacitor : 1.2.0
COCOAPODS: 1.6.1

Thank you

**Comment by zwacky** [comment]
I ran into the same issue.

here's my output:

```
2019-10-17 15:43:33.807722+0200 JustWatch[94147:3892889] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-17 15:43:33.807918+0200 JustWatch[94147:3892889] [Assert] Cannot be called with asCopy = NO on non-main thread.
[Youtube Player Plugin Native iOS]: YPViewController::viewDidLoad
2019-10-17 15:43:33.808235+0200 JustWatch[94147:3892889] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-17 15:43:33.808342+0200 JustWatch[94147:3892889] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-17 15:43:33.808518+0200 JustWatch[94147:3892889] *** Assertion failure in -[FBSSerialQueue assertOnQueue], /BuildRoot/Library/Caches/com.apple.xbs/Sources/FrontBoardServices_Sim/FrontBoard-619.2/FrontBoardServices/FBSSerialQueue.m:98
2019-10-17 15:43:33.811357+0200 JustWatch[94147:3892889] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'threading violation: expected the main thread'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23baa1ee __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff50864b20 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff23ba9f68 +[NSException raise:format:arguments:] + 88
	3   Foundation                          0x00007fff25614de9 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 191
	4   FrontBoardServices                  0x00007fff3652baeb -[FBSSerialQueue assertOnQueue] + 236
	5   FrontBoardServices                  0x00007fff364de8f5 -[FBSSceneImpl updateClientSettings:withTransitionContext:] + 70
	6   FrontBoardServices                  0x00007fff364deb40 -[FBSSceneImpl updateClientSettingsWithTransitionBlock:] + 154
	7   FrontBoardServices                  0x00007fff364dea65 -[FBSSceneImpl updateClientSettingsWithBlock:] + 110
	8   UIKitCore                           0x00007fff470caddc -[FBSScene(UIApp) updateUIClientSettingsWithBlock:] + 160
	9   UIKitCore                           0x00007fff46cc75cd -[_UISystemAppearanceManager updateScreenEdgesDeferringSystemGestures] + 374
	10  UIKitCore                           0x00007fff46f28c64 __70-[UIViewController setNeedsUpdateOfScreenEdgesDeferringSystemGestures]_block_invoke_2 + 118
	11  UIKitCore                           0x00007fff46f06f26 -[UIViewController _updateSystemAppearanceWithRecursionBlock:action:] + 295
	12  UIKitCore                           0x00007fff46f288ab -[UIViewController _setPresentedStatusBarViewController:] + 220
	13  UIKitCore                           0x00007fff46f18c17 -[UIViewController _presentViewController:modalSourceViewController:presentationController:animationController:interactionController:completion:] + 1381
	14  UIKitCore                           0x00007fff46f1a602 -[UIViewController _presentViewController:withAnimationController:completion:] + 4325
	15  UIKitCore                           0x00007fff46f1ce7b __63-[UIViewController _presentViewController:animated:completion:]_block_invoke + 98
	16  UIKitCore                           0x00007fff46f1d393 -[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 511
	17  UIKitCore                           0x00007fff46f1cdd9 -[UIViewController _presentViewController:animated:completion:] + 187
	18  UIKitCore                           0x00007fff46f1d040 -[UIViewController presentViewController:animated:completion:] + 150
	19  CapacitorYoutubePlayer              0x000000010359570d $s22CapacitorYoutubePlayer0bC0C10initializeyySo13CAPPluginCallCF + 4381
	20  CapacitorYoutubePlayer              0x0000000103595cb4 $s22CapacitorYoutubePlayer0bC0C10initializeyySo13CAPPluginCallCFTo + 68
	21  Capacitor                           0x00000001033aa2f3 $s9Capacitor9CAPBridgeC12handleJSCall4callyAA0D0C_tFyycfU_Tf2nnni_n + 739
	22  Capacitor                           0x00000001033823ed $sIeg_IeyB_TR + 45
	23  libdispatch.dylib                   0x00000001043c6df8 _dispatch_call_block_and_release + 12
	24  libdispatch.dylib                   0x00000001043c7d6c _dispatch_client_callout + 8
	25  libdispatch.dylib                   0x00000001043ce613 _dispatch_lane_serial_drain + 788
	26  libdispatch.dylib                   0x00000001043cf1a3 _dispatch_lane_invoke + 422
	27  libdispatch.dylib                   0x00000001043daa64 _dispatch_workloop_worker_thread + 719
	28  libsystem_pthread.dylib             0x00007fff518cd611 _pthread_wqthread + 421
	29  libsystem_pthread.dylib             0x00007fff518cd3fd start_wqthread + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by matheusdavidson** [comment]
also getting the same error

**Comment by KyDenZ** [comment]
In the end I use a cordova plug-in that works well with capacitor:

https://github.com/abomadi/CordovaYoutubeVideoPlayer

Conversation :
https://github.com/ionic-team/ionic/issues/19612

**Comment by abritopach** [comment]
Sorry for my late reply :( but I've changed jobs recently.

I've solved the problem. I updated the version and uploaded it to NPM (capacitor-youtube-player 1.0.0-alpha.1-19).

Now that I have more time, I'm going to keep working on the project.

If anyone has any suggestions, improvements or ideas to add or even constructive criticism are welcome.

I will go over the other issues these days.

Thanks.
